python/ray/dag/compiled_dag_node.py

```diff
@@ -669,6 +677,11 @@ def _get_node_id(self):
         input_task = self.idx_to_task[self.input_task_idx]
         # Register custom serializers for inputs provided to dag.execute().
         input_task.dag_node.type_hint.register_custom_serializer()
+        ctx = ChannelContext.get_current()
+        self.serialization_ctx = ctx.serialization_context
```

Comment:
`serialization_ctx` is a per-worker context. The reason we also set it as an attribute of `CompiledDag` is because that object lives on the driver, which runs different code from the actors participating in the DAG.

I cached it as an attribute of the actor/driver for convenience, but can remove this and directly look it up with `ChannelContext.get_current().serialization_context` for clarity.

Reply:
Actually for clarity, I'm going to move the call to `set_use_external_transport` directly into the channels.